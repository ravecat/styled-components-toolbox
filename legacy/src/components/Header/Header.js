import styled from "styled-components";

import { theme } from "common/theme";
import BaseComponent from "components/BaseComponent";

const Header = styled(BaseComponent).attrs(({ tag = "h1" }) => ({
  as: tag
}))`
  color: ${theme.primaryTextColor};
  font-size: ${({ size }) => size}px;
  font-style: ${({ italic }) => italic && `italic`};
  text-transform: ${({ uppercase }) => uppercase && `uppercase`};
  line-height: ${({ lheight }) => lheight || theme.lineHeight};
  font-weight: ${({ bold }) => bold && `700`};
  text-decoration: ${({ underline }) => underline && `underline`};
`;

export default Header;
