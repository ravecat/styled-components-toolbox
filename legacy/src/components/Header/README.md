## Header

An HTML heading, one of h1, h2, h3, h4, h5, h6.

### API

| Property    | Description                                 | Type    | Default |
| ----------- | ------------------------------------------- | ------- | ------- |
| _bold_      | set bold heading                            | boolean | `false` |
| _tag_       | set which HTML heading level should be used | string  | `h1`    |
| _uppercase_ | convert the heading to uppercase            | boolean | `false` |
