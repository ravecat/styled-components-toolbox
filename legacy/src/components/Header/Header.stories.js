import React from "react";
import { text, boolean, select } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Header from "./index";

const DEFAULT_TAG = "h1";
const OPTIONS = ["h1", "h2", "h3", "h4", "h5", "h6"];

storiesOf("Header", module).add("Default", () => (
  <Header
    bold={boolean("bold", false)}
    italic={boolean("italic", false)}
    lheight={select("lheight", [0.5, 1, 1.5, 2], 1)}
    size={select("size", [6, 10, 14, 18, 22, 26], 14)}
    tag={select("tag", OPTIONS, DEFAULT_TAG)}
    underline={boolean("underline", false)}
    uppercase={boolean("uppecase", false)}
  >
    {text("value", "Header")}
  </Header>
));
