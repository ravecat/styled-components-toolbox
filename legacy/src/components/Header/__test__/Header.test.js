import React from "react";
import renderer from "react-test-renderer";

import Header from "../index";

it("<Header> renders correctly", () => {
  const tree = renderer.create(<Header>Header</Header>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("<Header> renders correctly as 'h3' tag", () => {
  const tree = renderer.create(<Header tag={"h3"}>Header</Header>).toJSON();
  expect(tree).toMatchSnapshot();
});
