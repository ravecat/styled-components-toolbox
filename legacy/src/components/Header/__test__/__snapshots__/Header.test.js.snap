// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders correctly 1`] = `
<h1
  className="sc-bdVaJa sc-bwzfXH loEeXr"
>
  Header
</h1>
`;

exports[`<Header> renders correctly as 'h3' tag 1`] = `
<h3
  className="sc-bdVaJa sc-bwzfXH loEeXr"
>
  Header
</h3>
`;
